sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/SupportsPartitionManagement.java

```diff
@@ -106,10 +106,19 @@ void replacePartitionMetadata(
         throws UnsupportedOperationException;
 
     /**
-     * List the identifiers of all partitions that contains the ident in a table.
+     * List the identifiers of all partitions that have the ident prefix in a table.
      *
      * @param ident a prefix of partition identifier
      * @return an array of Identifiers for the partitions
      */
     InternalRow[] listPartitionIdentifiers(InternalRow ident);
+
+    /**
+     * List the identifiers of all partitions that match to the ident by names.
+     *
+     * @param names the names of partition values in the identifier.
+     * @param ident a partition identifier values.
+     * @return an array of Identifiers for the partitions
+     */
+    InternalRow[] listPartitionByNames(String[] names, InternalRow ident);
```

Comment:
Is a user supposed to access to `Table` directly? Looks a bit odd that `listPartitionByNames` interface is added but not used in the Spark internal side.

Reply:
Yea we should remove the old API as it's not released yet.

Reply:
The method `listPartitionIdentifiers()` is used only in `partitionExists()` and in tests. Let me remove it and replace its usage by `listPartitionByNames()`.

How about renaming `listPartitionByNames()` to just `listPartitions()`?

Reply:
> Looks a bit odd that listPartitionByNames interface is added but not used in the Spark internal side.

@HyukjinKwon `listPartitionByNames()` will be used by V2 commands like `SHOW PARTITIONS` and `SHOW TABLE EXTENDED`.

Reply:
> Yea we should remove the old API as it's not released yet.

Frankly speaking, I would remove `listPartitionIdentifiers()` separately as this requires unrelated changes to list partition by names.

Reply:
I think the final API name should still be `listPartitionIdentifiers`

Reply:
We can remove the old one in your next PR.

Reply:
> I think the final API name should still be `listPartitionIdentifiers`

Agree with this.